src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java

```diff
@@ -62,38 +76,50 @@ public static void addDefinitions(ConfigDef configDef) {
             "Set to 'true' to customize how an SSL connection is created. Only supported if securityContextType is 'BASIC' or 'DIGEST'.")
         .define(TLS_VERSION, Type.STRING, "TLSv1.2", Importance.LOW,
             "The TLS version to use for custom SSL")
-        .define(SSL_HOST_VERIFIER, Type.STRING, "ANY", Importance.LOW,
-            "The host verification strategy for custom SSL; either 'ANY', 'COMMON', or 'STRICT'")
+        .define(SSL_HOST_VERIFIER, Type.STRING, "ANY", SSL_HOST_VERIFIER_RV, Importance.LOW,
+            "The host verification strategy for custom SSL; either 'ANY', 'COMMON', or 'STRICT'",
+            null, -1, ConfigDef.Width.SHORT, SSL_HOST_VERIFIER, SSL_HOST_VERIFIER_RV)
         .define(SSL_MUTUAL_AUTH, Type.BOOLEAN, null, Importance.LOW,
             "Set this to true for 2-way SSL; defaults to 1-way SSL");
     }
 
     protected MarkLogicConfig(ConfigDef definition, Map<?, ?> originals, boolean doLog) {
         super(definition, originals, doLog);
     }
-}
 
-class CustomRecommenderAndValidator implements ConfigDef.Recommender, ConfigDef.Validator {
-    private List<Object> validValues;
+    public static class CustomRecommenderAndValidator implements ConfigDef.Recommender, ConfigDef.Validator {
 
-    public CustomRecommenderAndValidator(String... validValues) {
-        this.validValues = Arrays.asList((Object[]) validValues);
-    }
+        private List<Object> validValues;
 
-    @Override
-    public List<Object> validValues(String name, Map<String, Object> parsedConfig) {
-        return validValues;
-    }
+        public CustomRecommenderAndValidator(String... validValues) {
+            this.validValues = Arrays.asList(validValues);
+        }
 
-    @Override
-    public boolean visible(String name, Map<String, Object> parsedConfig) {
-        return true;
-    }
+        @Override
+        public List<Object> validValues(String name, Map<String, Object> parsedConfig) {
+            return validValues;
+        }
+
+        @Override
+        public boolean visible(String name, Map<String, Object> parsedConfig) {
+            return true;
+        }
 
-    @Override
-    public void ensureValid(String name, Object value) {
-        if (!validValues.contains(value)) {
-            throw new ConfigException("Invalid value: " + value + "; must be one of: " + validValues);
+        @Override
+        public void ensureValid(String name, Object value) {
+            if(name.equals(MarkLogicConfig.CONNECTION_HOST) || name.equals(MarkLogicConfig.CONNECTION_PORT)) {
+                if(value == null || value.toString().isEmpty()) {
+                    throw new ConfigException(String.format("%s cannot be empty", name));
+                }
```

Comment:
Using the Validators that are included in the Kafka library will remove the necessity for this section of code.